atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ProfilingKeyValueService.java

```diff
@@ -390,7 +404,66 @@ public RowColumnRangeIterator getRowsColumnRange(TableReference tableRef,
                         LoggingArgs.durationMillis(stopwatch)));
     }
 
-    private  <T> T maybeLog(Supplier<T> action, BiConsumer<LoggingFunction, Stopwatch> logger) {
+    private RowColumnRangeIterator wrapRowColumnRangeIteratorWithLogger(
+            Supplier<RowColumnRangeIterator> rowColumnRangeIteratorSupplier,
+            BiConsumer<LoggingFunction, Stopwatch> logger) {
+        RowColumnRangeIterator rowColumnRangeIterator = rowColumnRangeIteratorSupplier.get();
```

Comment:
don't think you need to pass a supplier here. can probably pass the iterator itself

Reply:
Done.